blockauth/apple/notification_service.py

```
@@ -55,6 +62,73 @@ def _parse_events_claim(raw: Any) -> dict[str, Any]:
+
+def _future_leeway_seconds() -> int:
+    return _int_setting("APPLE_NOTIFICATION_FUTURE_LEEWAY_SECONDS", 60, minimum=0)
+
+
+def _validate_event_time_fresh(events: dict[str, Any]) -> None:
+    event_time = _event_time_seconds(events.get("event_time"))
+    now = time.time()
+    max_age = _max_age_seconds()
+    leeway = _future_leeway_seconds()
+    if event_time < now - max_age:
+        raise AppleNotificationVerificationFailed("Apple notification event_time is stale")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)